pkg/cache/resource_node.go

```diff
@@ -154,6 +154,7 @@ func (r ResourceNode) calculateLendable() map[corev1.ResourceName]int64 {
 }
 
 func updateClusterQueueResourceNode(cq *clusterQueue) {
+	cq.AllocatableResourceGeneration += 1
```

Comment:
I see, so when updating any CQ (see [here](https://github.com/kubernetes-sigs/kueue/blob/afb49468223ca81419b13e45923f752f803dd80e/pkg/cache/clusterqueue.go#L136-L149)) we either call `updateClusterQueueResourceNode` directly, or call it for each child via `updateCohortResourceNode`. This way any CQ update entails generation update of all CQs in the cohort.